monai/transforms/io/dictionary.py

```diff
@@ -127,22 +133,48 @@ def __call__(self, data, reader: Optional[ImageReader] = None):
 
         """
         d = dict(data)
+        image_shapes = []
+        # From python 3.6 and up, dictionaries are ordered, so we can assume that the dict orders we get
+        # are all in the same order, unless the user goes out of their way to shuffle them.
+        first_path = list(d.values())[0]
+
         for key, meta_key, meta_key_postfix in self.key_iterator(d, self.meta_keys, self.meta_key_postfix):
-            data = self._loader(d[key], reader)
+            image_data = self._loader(d[key], reader)
```

Comment:
Could you revert the naming of `image_data` back to `data`? Would simplify the `diff`.

Reply:
Done, I wanted to keep the input `data` intact in case I needed to log it in the warning, I've just gone and stored a copy in `input_dict` instead.